pkg/api/pluginproxy/access_token_provider_test.go

```diff
@@ -91,4 +98,172 @@ func TestAccessToken(t *testing.T) {
 			So(token2, ShouldEqual, "abc")
 		})
 	})
+
+	Convey("Plugin with token auth route", t, func() {
+		apiHandler := http.NewServeMux()
+		server := httptest.NewServer(apiHandler)
+
+		defer server.Close()
+		pluginRoute := &plugins.AppPluginRoute{
+			Path:   "pathwithtokenauth1",
+			Url:    "",
+			Method: "GET",
+			TokenAuth: &plugins.JwtTokenAuth{
+				Url: server.URL + "/oauth/token",
+				Scopes: []string{
+					"https://www.testapi.com/auth/monitoring.read",
+					"https://www.testapi.com/auth/cloudplatformprojects.readonly",
+				},
+				Params: map[string]string{
+					"grant_type":    "client_credentials",
+					"client_id":     "{{.JsonData.client_id}}",
+					"client_secret": "{{.SecureJsonData.client_secret}}",
+					"audience":      "{{.JsonData.audience}}",
+					"client_name":   "datasource_plugin",
+				},
+			},
+		}
+
+		templateData := templateData{
+			JsonData: map[string]interface{}{
+				"client_id": "my_client_id",
+				"audience":  "www.example.com",
+			},
+			SecureJsonData: map[string]string{
+				"client_secret": "my_secret",
+			},
+		}
+
+		var authCalls int
+		apiHandler.HandleFunc("/oauth/token", func(w http.ResponseWriter, req *http.Request) {
+			err := json.NewEncoder(w).Encode(token)
+			if err != nil {
+				t.Error(err)
+			}
+			authCalls++
+		})
+
+		Convey("Should parse token, with different fields and types", func() {
+			type tokenTestDescription struct {
+				desc              string
+				expiresIn         interface{}
+				expiresOn         interface{}
+				expectedExpiresOn int64
+			}
+
+			mockTimeNow(time.Now().Unix())
+			defer resetTimeNow()
+			provider := newAccessTokenProvider(&models.DataSource{}, pluginRoute)
+
+			testCases := []tokenTestDescription{
+				{
+					desc:              "token with expires_in in string format",
+					expiresIn:         "3600",
+					expiresOn:         nil,
+					expectedExpiresOn: timeNow().Unix() + 3600,
+				},
+				{
+					desc:              "token with expires_in in int format",
+					expiresIn:         5200,
```

Comment:
I would suggest to use `3600` as a value here so as to be clearer the difference with the above case.